jni/src/faiss_wrapper.cpp

```diff
@@ -81,6 +85,85 @@ bool isIndexIVFPQL2(faiss::Index * index);
 // IndexIDMap which has member that will point to underlying index that stores the data
 faiss::IndexIVFPQ * extractIVFPQIndex(faiss::Index * index);
 
+void knn_jni::faiss_wrapper::CreateBinaryIndex(knn_jni::JNIUtilInterface * jniUtil, JNIEnv * env, jintArray idsJ, jlong vectorsAddressJ, jint dimJ,
+                        jstring indexPathJ, jobject parametersJ) {
+    if (idsJ == nullptr) {
+        throw std::runtime_error("IDs cannot be null");
+    }
+
+    if (vectorsAddressJ <= 0) {
+        throw std::runtime_error("VectorsAddress cannot be less than 0");
+    }
+
+    if(dimJ <= 0) {
+        throw std::runtime_error("Vectors dimensions cannot be less than or equal to 0");
+    }
+
+    if (indexPathJ == nullptr) {
+        throw std::runtime_error("Index path cannot be null");
+    }
+
+    if (parametersJ == nullptr) {
+        throw std::runtime_error("Parameters cannot be null");
+    }
```

Comment:
We should abstract the common code in separate functions. There is a lot of copy paste. Lets abstract things out.

Reply:
@navneet1v Will reduce duplicated logic.
@shatejas As I mentioned in other comment, If there should be a class for index creation related logic, it should be in a java side. Adding more logic in JNI side will make it harder to maintain.

Reply:
@shatejas On a second thought, maybe having a factory class might be good ad you suggested...